core/vm/lfvm/instructions.go

```diff
@@ -764,35 +906,51 @@ func opCall(c *context) {
 	toAddr := common.Address(addr.Bytes20())
 
 	// Charge for transfering value to a new address
-	if !value.IsZero() && !c.stateDB.Exist(toAddr) && !c.UseGas(params.CallNewAccountGas) {
-		return
+	if !value.IsZero() {
+		base_gas += params.CallValueTransferGas
 	}
 
-	// Charge value-transfere fees
-	if !value.IsZero() && !c.UseGas(params.CallValueTransferGas) {
-		return
+	// if evm.chainRules.IsEIP158 according to GETH it is EIP158 since 2016
+	// !!!! but need to touch stateDB for the address to have it in the substate record key/value
+	if !value.IsZero() && !c.stateDB.Exist(toAddr) {
+		base_gas += params.CallNewAccountGas
 	}
 
-	if !c.UseGas(gas) {
-		return
+	cost := callGas(c.contract.Gas, base_gas, provided_gas)
+
+	if warmAccess {
+		if !c.UseGas(base_gas + cost) {
+			return
+		}
+	} else {
+		// In case of a cold access, we temporarily add the cold charge back, and also
+		// add it to the returned gas. By adding it to the return, it will be charged
+		// outside of this function, as part of the dynamic gas, and that will make it
+		// also become correctly reported to tracers.
+		c.contract.Gas += coldCost
+		if !c.UseGas(base_gas + cost + coldCost) {
+			return
+		}
 	}
 
-	// Get the arguments from the memory.
-	args := c.memory.GetSlice(int64(inOffset.Uint64()), int64(inSize.Uint64()))
+	// first use static and dynamic gas cost and then resize the memory
+	// when out of gas is happening, then mem should not be resized
+	c.memory.EnsureCapacityWithoutGas(needed_memory_size, c)
 
 	var bigVal = big0
 	//TODO: use uint256.Int instead of converting with toBig()
```

Comment:
Can we raise the TODOs as issues in GitHub as well? That might be easier to keep track of the work.